crates/solidity/testing/perf/cargo/src/compilation_builder.rs

```diff
@@ -0,0 +1,49 @@
+use anyhow::Result;
+use semver::{BuildMetadata, Prerelease};
+use slang_solidity::compilation::{CompilationBuilder, CompilationBuilderConfig, CompilationUnit};
+
+use crate::dataset::SolidityProject;
+
+struct ProjectConfig<'a> {
+    project: &'a SolidityProject,
+}
+
+impl CompilationBuilderConfig for ProjectConfig<'_> {
```

Comment:
Nit: I only now realized this, but you can implement `CompilationBuilderConfig` for `SolidityProject` directly, so you don't need a `ProjectConfig` and `file_contents` and `resolve_import` can live in this module.